src/pipeline/media-decode.js

```
@@ -3,131 +3,201 @@
+
+    // 3. Ring-buffer via ScriptProcessorNode.
+    //    SPN is deprecated but universally supported and gives us synchronous
+    //    PCM access — the only option for a MediaElementSource on a live ctx.
+    const numChannels = 2;
+    const estimatedFrames = Math.ceil(duration * SAMPLE_RATE) + SPN_BLOCK_SIZE * 4;
+    const chunks = [];            // Array<Array<Float32Array>>  [block][ch]
+    let capturedFrames = 0;
+    let resolveDone, rejectDone;
+    const donePromise = new Promise((res, rej) => { resolveDone = res; rejectDone = rej; });
+
+    // eslint-disable-next-line no-deprecated
```

ESLint

[error] Definition for rule 'no-deprecated' was not found.

(no-deprecated)